### PR TITLE
[Google Sheets] Preserve the columns' source order on saving

### PR DESCRIPTION
**Pull request by DmitrySharabin**

### CODE REVIEW

File: backends/google/sheets/google-sheets.js [hunk 1]
```diff
@@ -138,8 +138,24 @@ export default class GoogleSheets extends Google {
...
-			data = data.map(obj => Object.values(obj));
+			// We must preserve the columns' source order.
+			data = data.map(obj => {
+				const ret = [];
+
+				for (const key of Object.keys(obj)) {
+					const index = columnNumbers.get(key) ?? key; // Why “?? key”? Handle the case when object keys are already column indices (e.g., when keys: []).
+
+					// FIXME: What if objects have “new” keys (i.e., index is not a number), e.g., the user wants to add new columns with data?
+					// What should we do? Should we add them at the end of the corresponding row? Will ret.push(obj[key]) do?
```
Comment (LeaVerou) [inline backends/google/sheets/google-sheets.js:153]: What currently happens? We should definitely not *lose* data. Adding them to the end seems fine.

Comment (DmitrySharabin) [reply backends/google/sheets/google-sheets.js:153]: I'm afraid, currently, nothing is happening. So we have another bug that should be fixed. I'll add new data to the end in the “order of appearance” inside of an object then.

**Review by LeaVerou — APPROVED** [review]

**Comment by DmitrySharabin** [comment]
@LeaVerou, changes from this PR are also gone. 😳

**Comment by LeaVerou** [comment]
Wow, this is freaky

**Pull request merged**